iree/compiler/Conversion/LinalgToSPIRV/Passes.cpp

```diff
@@ -166,6 +166,10 @@ void buildSPIRVTransformPassPipeline(OpPassManager &pm,
   pm.nest<ModuleOp>().addPass(createInlinerPass());
 
   if (options.usingLinalgOnTensors) {
+    // flow.dispatch.workgroups performed abstract tiling and distribution. Make
+    // them concrete now since we know the target and settings now.
+    pm.addPass(createConcretizeTileAmongWorkgroupsPass());
```

Comment:
THis is running way to early. Could we run this in the addLinalgToSPIRV passes above. I am confused cause the `LinalgTileAndFusePass` is basically looking at the LaunchConfig and then setting the tile sizes as well as setting the entry point region for number of workgroups. THis seems like you want to do it there. All of these passes are actually common between CPU and GPU and I dont understand why we need to concretize this information here.

Reply:
Actually looking at what you are trying to do, it seems like it might just be better to split the LinalgTileAndFuse pass into a  `LinalgTileAndDistributePass` and a `materializeGPULaunchConfigurationPass` and invoke the former for the legacy pass and the latter for the linalg on tensors path (like it is done here). All the patterns you want go right into the `materializeGPULaunchConfigurationPass`. 

You can literally copy paste the `LinalgTileAndDistributePass` and change it to use `initGPULaunchConfig` instead of `initCPULaunchConfig`.  (Same for `materializeGPULaunchConfigurationPass` which is a copy + modify of `materializeCPULaunchConfigurationPass`). This is intentional. We can then unify the two by having a pass that materializes the configuration as an attribute on the op and then you can use the same pass for both.

Reply:
Re: running early. I placed it there b/c flow dispatch region formation is abstract tiling and distribution while this pass is concretizing that. So it logically connects to me. I'm fine moving to a later stage too. 

Re: passes and their relations. Yeah I was previously planning to do the clean up involving `LinalgTileAndFusePass` after this patch. But given you are asking about it, done it in this patch then. Now `ConcretizeTileAmongWorkgroups` handles supplying the concrete first-level tiling configurations for linalg on tensors path. A new `LinalgTileAndDistributePass` is created for tiling and distributing linalg on buffer ops for the first-level (and it will be deleted later). `LinalgTileAndFusePass` only handles tiling and vectorization inside a single workgroup now. (It needs to be renamed to something like `TileAndVectorizeInOneWorkgroup` but that can happen later.)